### DISCUSSION TITLE
Failsafe against broken updates?

### DISCUSSION THREAD

**Discussion by iamcalledrob** [sparkle-project/Sparkle#2387] [Q&A] [answered]
Hi Sparkle folks

There's always a chance that an app will crash on launch on an end user's machine--try as we might, bugs will always make it into production, and sometimes only surface "in the wild".

I'm curious how to protect against broken updates like this. It's my understanding that Sparkle's update checking operates in-process with the app. So if someone were to install an update to a broken build, they'd be unable to update out of that state when I push a fix.

I've scoured the docs and don't see any advice for situation. Does Sparkle offer any failsafe? Are there any workarounds, e.g. some sort of helper app?

(Apologies in advance if this isn't the right place to post)

**Comment by zorgiepoo** [comment, marked as answer]
Not really if the reason your app is that unstable is unrelated to Sparkle. Sparkle may have a fail safe if the updating process of your app is broken for example (usually due to invalid configuration) which may involve posting an informational only update. The best you can do in the former case is to harden on your testing before releasing and learning from past mistake and how to prevent those going forward.

**Reply by iamcalledrob** [reply]
Thanks for confirming. Will look into other solutions.

The issue here is that even with decent testing there's always a chance of accidentally launching with an insta-crash (for example, perhaps there's an incompatibility between the update and a dot release of macos).
